casr/src/bin/casr-lua.rs

```diff
@@ -0,0 +1,155 @@
+                .help("Stdin file for program"),
+        )
+        .arg(
+            Arg::new("timeout")
+                .short('t')
+                .long("timeout")
+                .action(ArgAction::Set)
+                .default_value("0")
+                .value_name("SECONDS")
+                .help("Timeout (in seconds) for target execution, 0 value means that timeout is disabled")
+                .value_parser(clap::value_parser!(u64))
+        )
+        .arg(
+            Arg::new("ignore")
+                .long("ignore")
+                .action(ArgAction::Set)
+                .value_parser(clap::value_parser!(PathBuf))
+                .value_name("FILE")
+                .help("File with regular expressions for functions and file paths that should be ignored"),
+        )
+        .arg(
+            Arg::new("strip-path")
+                .long("strip-path")
+                .env("CASR_STRIP_PATH")
+                .action(ArgAction::Set)
+                .value_name("PREFIX")
+                .help("Path prefix to strip from stacktrace"),
+        )
+        .arg(
+            Arg::new("ARGS")
+                .action(ArgAction::Set)
+                .num_args(1..)
+                .last(true)
+                .required(true)
+                .help("Add \"-- <path> <arguments>\" to run"),
+        )
+        .get_matches();
+
+    init_ignored_frames!("lua");
+    if let Some(path) = matches.get_one::<PathBuf>("ignore") {
+        util::add_custom_ignored_frames(path)?;
+    }
+    // Get program args.
+    let argv: Vec<&str> = if let Some(argvs) = matches.get_many::<String>("ARGS") {
+        argvs.map(|s| s.as_str()).collect()
+    } else {
+        bail!("Wrong arguments for starting program");
+    };
+
+    // Get stdin for target program.
+    let stdin_file = util::stdin_from_matches(&matches)?;
+
+    // Get timeout
+    let timeout = *matches.get_one::<u64>("timeout").unwrap();
+
+    // Run program.
+    let mut cmd = Command::new(argv[0]);
+    if let Some(ref file) = stdin_file {
+        cmd.stdin(std::fs::File::open(file)?);
+    }
+    if argv.len() > 1 {
+        cmd.args(&argv[1..]);
+    }
+    let result = util::get_output(&mut cmd, timeout, true)?;
+    let stderr = String::from_utf8_lossy(&result.stderr);
+
+    // Create report.
+    let mut report = CrashReport::new();
+    report.executable_path = argv[0].to_string();
+    if argv.len() > 1 {
+        if let Some(fname) = Path::new(argv[0]).file_name() {
+            let fname = fname.to_string_lossy();
+            if fname.starts_with("lua") && !fname.ends_with(".lua") && argv[1].ends_with(".lua") {
+                report.executable_path = argv[1].to_string();
+            }
+        }
+    }
+    report.proc_cmdline = argv.join(" ");
+    let _ = report.add_os_info();
+    let _ = report.add_proc_environ();
+
+    // Extract lua exception
+    let Some(exception) = LuaException::new(&stderr) else {
+        bail!("Lua exception is not found!");
```

Comment:
imagine a Lua code like the following:

```lua
[0] ~/sources/casr $ cat exc6.lua 
#!/bin/env lua

function a()
  error("Boom!", 3); 
end

function b() a(); end

function c() b(); end

function d() c(); end

function e() d(); end

function f() e(); end

function errhandler(msg)
  print (msg .. "\nstack traceback: \n" .. debug.traceback(2, 3, 2));
end

xpcall(f, errhandler);
```

that produces an output:

```
[0] ~/sources/casr $ ./exc6.lua 
./exc6.lua:9: Boom!
stack traceback: 
2
stack traceback:
        ./exc6.lua:4: in function 'a'
        ./exc6.lua:7: in function 'b'
        ./exc6.lua:9: in function 'c'
        ./exc6.lua:11: in function 'd'
        ./exc6.lua:13: in function 'e'
        ./exc6.lua:15: in function <./exc6.lua:15>
        [C]: in function 'xpcall'
        ./exc6.lua:21: in main chunk
        [C]: in ?
```

`casr-lua` will not find traceback there:

```
$ ./target/debug/casr-lua -o lua.casrep -- ./exc6.lua 
Error: Lua exception is not found!
```

Reply:
The error occurs only due to extra spaces after traceback

Reply:
We can think about storing custom (maybe multiline) error as `Explanation` or `Description` fields, but it's not clear how to extract `ShortDescription` or etc

Reply:
Fixed